modules/user_accounts/php/edit_user.class.inc

```diff
@@ -323,13 +325,63 @@ class Edit_User extends \NDB_Form
                     $ex_pending = $v;
                 }
             }
+            foreach ($ex_curr_sites as $k => $v) {
+                unset($values[$k]);
+            }
+        }
+        unset($values['examiner_radiologist']);
+        unset($values['examiner_pending']);
+        // END SETUP EXAMINER VALUES
+
+        // make the set
+        $set = array();
+        foreach ($values as $key => $value) {
+            // Password updates are handled separately
+            if (!empty($value) && $key != 'Password_hash') {
+                $set[$key] = $value;
+            } else {
+                $set[$key] = null;
+            }
+        }
+        // update the user
+        if ($this->isCreatingNewUser()) {
+            // insert a new user
+            \User::insert($set);
+            $user = \User::factory($set['UserID']);
+            $uid  = $user->getData('ID');
+            foreach ($us_curr_sites as $site) {
+                $DB->insert(
+                    'user_psc_rel',
+                    array(
+                        "UserID"   => $uid,
+                        "CenterID" => $site,
+                    )
+                );
+            }
+            foreach ($us_curr_projects as $project) {
+                $DB->insert(
+                    'user_project_rel',
+                    array(
+                        "UserID"    => $uid,
+                        "ProjectID" => $project,
```

Comment:
@CamilleBeau I'm not a fan of the examiner update section being split into 2 with this section (unrelated to examiners) now being in the middle. Can you instead move lines 303-334 to after 376 ?

Reply:
The problem here is that there is some processing of the $values array that is done in those lines, which is used in the insert for the new user. However, we also need the insert for the new user to happen before examiner update because we need the userID to be generated beforehand. One option would be to hand-pick each of the values from $values that will be used in inserting the user instead of unsetting the values. That way all of the examiner related code could be after the insert. This might be bulky code though.

Reply:
i c